packages/itwin/property-grid/src/components/PropertyGrid.tsx

```diff
@@ -322,11 +327,25 @@ export const PropertyGrid = ({
 
         setContextMenuItemInfos(items.length > 0 ? items : undefined);
       }
+
+      // Fcn for updating toggle for Hide / Show Empty Fields menu options
+      const updateShowNullValues = async (value: boolean) => {
```

Comment:
move this out of the buildContextMenu routine and have it use the `useCallback`, basically it should be idential setup to the `onHideNull`/`onShowNull` except that it's combing their logic and additionally persisting the value